### DISCUSSION TITLE
Adding connection using the cli

### DISCUSSION THREAD

**Discussion by FraFTri** [apache/airflow#15984] [Q&A]
**Apache Airflow version**: v2.0.1

**What happened**:

I want to add connections using the cli

`airflow connections add 'example_path_b' --conn-uri 'fs://?path=%2Fopt%2Fairflow%2Fdags%2Fb'`
Error: 
`The URI provided to --conn-uri is invalid: fs://?path=%2Fopt%2Fairflow%2Fdags%2Fb`

Adding this Connection with the UI works.
airflow connections get example_path_a

<pre>airflow@cc3f4165ec88:/$ airflow connections get example_path_a
<b>id </b>|<b> conn_id        </b>|<b> conn_type </b>|<b> description </b>|<b> host </b>|<b> schema </b>|<b> login </b>|<b> password </b>|<b> port </b>|<b> is_encrypted </b>|<b> is_extra_encrypted </b>|<b> extra_dejson                    </b>|<b> get_uri                               </b>
===+================+===========+=============+======+========+=======+==========+======+==============+====================+=================================+=======================================
50 | example_path_a | fs        |             |      |        |       | None     | None | False        | False              | {&apos;path&apos;: &apos;/opt/airflow/dags/a&apos;} | fs://?path=%2Fopt%2Fairflow%2Fdags%2Fa
<font color="#AAAAAA"><i>                                                                                                                                                                                                      </i></font>
airflow@cc3f4165ec88:/$ airflow connections add &apos;example_path_b&apos; --conn-uri &apos;fs://?path=%2Fopt%2Fairflow%2Fdags%2Fb&apos;
The URI provided to --conn-uri is invalid: fs://?path=%2Fopt%2Fairflow%2Fdags%2Fb
</pre>

I tried to copy the uri from `airflow connection get ...`

I also tried to generate the connection URI with the methods provided in the docs.
https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html

**How to reproduce it**:

Add a new connection with the UI.

1. Enter the Conn id (example_path_a)
2. Click on File (path)
2. Enter a JSON at the Extra field: {"path":"/opt/airflow/dags/a"}
3. Save the connection
4. Get the uri via: `airflow connections get example_path_a`
5. Try to add this connection with `airflow connections add example_path_a --conn-uri 'uri from step4'`

What is wrong with my uri? Should this be reported as a bug?

**Comment by ephraimbuddy** [comment]
Can you try and generate the connection URI as seen here: https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#generating-a-connection-uri

**Reply by FraFTri** [reply]
Thank you for your help. I already tried to generate the URI with the Connection class and get_uri() method. 
It results ins the same uri with the corresponding error.
Maybe some of the values may not be empty using the CLI?

**Reply by ephraimbuddy** [reply]
Oh sorry for the stress.
For CLI you need to use a colon between login and password like this:
`airflow connections add example_path_d --conn-uri 'fs://:?path=opt%2Fairflow%2Fdags%2Fa'`
The above worked for me on airflow 2.0.2. 
Looks like a bug though
